tests/capi/unittest_capi_service_training_offloading.cc

```diff
@@ -0,0 +1,292 @@
+#endif
+
+    /* should start receiver */
+    g_autofree gchar *receiver_config
+        = _get_config_path ("training_offloading_receiver.conf");
+    status = ml_service_new (receiver_config, &client_h);
+    ASSERT_EQ (status, ML_ERROR_NONE);
+
+    g_autofree gchar *sender_config = _get_config_path ("training_offloading_sender.conf");
+    status = ml_service_new (sender_config, &server_h);
+    ASSERT_EQ (status, ML_ERROR_NONE);
+  }
+
+  /**
+   * @brief Teardown method for each test case.
+   */
+  void TearDown () override
+  {
+    if (dbus) {
+      g_test_dbus_down (dbus);
+      g_object_unref (dbus);
+    }
+
+    status = ml_service_destroy (client_h);
+    EXPECT_EQ (ML_ERROR_NONE, status);
+
+    status = ml_service_destroy (server_h);
+    EXPECT_EQ (ML_ERROR_NONE, status);
+  }
+
+  /**
+   * @brief Get available port number.
+   */
+  static guint _get_available_port (void)
+  {
+    struct sockaddr_in sin;
+    guint port = 0;
+    gint sock;
+    socklen_t len = sizeof (struct sockaddr);
+
+    sin.sin_family = AF_INET;
+    sin.sin_addr.s_addr = INADDR_ANY;
+    sin.sin_port = htons (0);
+
+    sock = socket (AF_INET, SOCK_STREAM, 0);
+    EXPECT_TRUE (sock > 0);
+    if (sock < 0)
+      return 0;
+
+    if (bind (sock, (struct sockaddr *) &sin, sizeof (struct sockaddr)) == 0) {
+      if (getsockname (sock, (struct sockaddr *) &sin, &len) == 0) {
+        port = ntohs (sin.sin_port);
+      }
+    }
+    close (sock);
+
+    EXPECT_TRUE (port > 0);
+    return port;
+  }
+};
+
+
+/**
+ * @brief Callback function for reply test.
+ */
+static void
+_receive_trained_model_cb (ml_service_event_e event, ml_information_h event_data, void *user_data)
+{
+  switch ((int) event) {
+
+    case ML_SERVICE_EVENT_REPLY:
+      {
+        g_warning ("received trained model");
+        break;
+      }
+    default:
+      break;
+  }
+}
+
+
+/**
+ * @brief Callback function for sink node.
+ */
+static void
+_sink_register_cb (ml_service_event_e event, ml_information_h event_data, void *user_data)
+{
+  ml_tensors_data_h data = NULL;
+  double *output = NULL;
+  size_t data_size = 0;
+  int status;
+  double result_data[4];
+  char *output_node_name = NULL;
+
+  switch (event) {
+    case ML_SERVICE_EVENT_NEW_DATA:
+      ASSERT_TRUE (event_data != NULL);
+
+      status = ml_information_get (event_data, "name", (void **) &output_node_name);
+      EXPECT_EQ (status, ML_ERROR_NONE);
+      EXPECT_STREQ (output_node_name, "training_result");
+
+      status = ml_information_get (event_data, "data", &data);
+      EXPECT_EQ (status, ML_ERROR_NONE);
+
+      status = ml_tensors_data_get_tensor_data (data, 0, (void **) &output, &data_size);
+      EXPECT_EQ (status, ML_ERROR_NONE);
+    default:
+      break;
+  }
+
+  if (output) {
+    for (int i = 0; i < 4; i++)
+      result_data[i] = output[i];
+    g_warning ("######## %s : %s: %d, name:%s >> [training_loss: %f, training_accuracy: %f, validation_loss: %f, validation_accuracy: %f]",
```

Comment:
Please change log level. debug or info.

Reply:
How about keeping the warning for later debugging?